stock_warehouse_analytic_purchase/models/purchase_order_line.py

```diff
@@ -0,0 +1,16 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import _, models, api
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    @api.depends('order_id.picking_type_id.warehouse_id.account_analytic_id', 'order_id.picking_type_id.warehouse_id.account_analytic_tag_ids')
+    def _compute_account_analytic_id(self):
+        super()._compute_account_analytic_id()
+        for record in self.filtered(lambda x: x.order_id.picking_type_id.warehouse_id.account_analytic_id):
```

Comment:
According to original method definition (https://github.com/odoo/odoo/blob/f2765d2cab5671a010404c36842bf1b4c4d6350b/addons/purchase/models/purchase.py#L1013) those special purchase lines (section and notes, not regular lines) should be ignored in the process